pkg/prober/prober.go

```diff
@@ -58,13 +57,13 @@ func DoProbe(probe Probe, probeStatus *ProbeStatus, initial bool) error {
 			tlsConfig.Certificates = []tls.Certificate{clientCert}
 		}
 
-		caCertPool, err := x509.SystemCertPool()
+		caCertPool, err := GetSystemCertPool(probe.Name)
```

Comment:
added a nil check

Reply:
I don't see a `nil`-check here.

It is possible for `GetSystemCertPool` to return `nil, error` in many cases, both for `unix` and `windows`. If this happens, we are only logging the error and continuing. Which means that `caCertPool` will be `nil` as we continue lower. This may cause later calls (like the on on line 71) to panic.